tests/pact/tc.repl

```diff
@@ -335,3 +335,36 @@
  "rtc object lists"
  1
  (rtc-object-list (let ((a {'a:1}) (b {'a: 2})) [a b])))
+
+;; A module with incorrect annotations for a function's return type.
+;; Runtime typechecking for function return types was added in
+;; pact-4.7.
+(module tc-test-function-return-types gov
+  (defcap gov () true)
+  (defun f:string (a: string) (+ 1 2))
+  (defun g:integer (a: integer) a))
+
+;; Functions with incompatible type annotations should not compose.
+(expect-failure
+  "composing type-incompatible, body-incompatible fails evaluation after pact-4.6"
+  (tc-test-function-return-types.g (tc-test-function-return-types.f ""))
+  )
+
+(expect-failure
+  "`compose` on type-incompatible, body-incompatible functions fails evalution after pact-4.6"
+  (compose (tc-test-function-return-types.f) (tc-test-function-return-types.g) ""))
```

Comment:
I don't think we need `compose` here either, compose simply applies each function, it doesn't perform any extra type checking.

Reply:
That's fair